hs_core/views/__init__.py

```diff
@@ -276,14 +276,7 @@ def add_metadata_element(request, shortkey, element_name, *args, **kwargs):
         # seems the user wants to delete all keywords - no need for pre-check in signal handler
         res.metadata.subjects.all().delete()
         is_add_success = True
-        if not res.can_be_public_or_discoverable:
-            res.raccess.public = False
-            res.raccess.discoverable = False
-            res.raccess.save()
-        elif not res.raccess.discoverable:
-            res.raccess.discoverable = True
-            res.raccess.save()
-
+        res.update_public_and_discoverable()
```

Comment:
@pkdash Some of these don't even seem to set resource_modified after modifying a resource....
It would seem to me that whenever one modifies a resource, a check should be made....